packages/nimble-components/src/table/types.ts

```diff
@@ -61,3 +62,24 @@ export interface TableActionMenuToggleEventDetail {
     recordIds: string[];
     columnId?: string;
 }
+
+/**
+ * The possible directions a table column can be sorted in.
+ */
+export const TableColumnSortDirection = {
+    none: undefined,
+    ascending: 'ascending',
+    descending: 'descending'
+} as const;
+export type TableColumnSortDirection =
+    (typeof TableColumnSortDirection)[keyof typeof TableColumnSortDirection];
+
+/**
+ * The possible operations to use when sorting a table column.
+ */
+export const TableColumnSortOperation = {
+    basic: 'basic',
```

Comment:
Could you add docs that describe the behavior of the sort operations? Inline docs here would be fine.

We could also consider documenting it in storybook but if this is only used by custom columns then we could also defer this until #976 .

Reply:
I added inline docs.